src/main/java/org/jacp/service/CompetitionService.java

```diff
@@ -27,7 +29,23 @@ public List<CompetitionEntity> getAllByStatus(Status status) {
                 .orElseThrow(() -> new NoSuchElementException("Competition with " + status + " status are not found"));
     }
 
-    public CompetitionEntity getCompetitionEntity(Long id){
+    public CompetitionEntity getCompetitionEntity(Long id) {
         return competitionRepository.getCompetitionEntityById(id);
     }
+
+    public CompetitionEntity startCompetition(Long id) {
+        CompetitionEntity competitionEntity = getCompetitionEntity(id);
+        competitionEntity.setStartDate(Date.from(Instant.now()));
+        competitionEntity.calculateEndDate();
+        competitionEntity.setStatus(Status.QUEUED);
+        create(competitionEntity);
```

Comment:
You don't need to call save() method here. Just annotate the method as transactional.

Reply:
Fixed